### PR TITLE
Run CI on Java 17

### PR DESCRIPTION
**Pull request by TimvdLippe**
Let's see how far we can get.

### CODE REVIEW

File: src/test/java/org/mockito/internal/creation/bytebuddy/SubclassByteBuddyMockMakerTest.java [hunk 1]
```diff
@@ -34,6 +35,7 @@ public void is_type_mockable_excludes_primitive_wrapper_classes() {
     }
 
     @Test
+    @Ignore("Broken on JDK 17")
```
Comment (TimvdLippe) [inline src/test/java/org/mockito/internal/creation/bytebuddy/SubclassByteBuddyMockMakerTest.java:38]: We were only running this on 17, but it seems to be failing anyways 😢

**Comment by TimvdLippe** [comment]
The CodeCov diff now shows we are correctly using the Java 16+ codepath for invoking the relevant proxy method in the proxy mockmaker. So everything is working as expected and we have more code coverage 🎉

**Comment by TimvdLippe** [comment]
@raphw Do you mind taking a look at the disabled test and let me know if that is intended?

File: src/test/java/org/mockito/internal/creation/bytebuddy/SubclassByteBuddyMockMakerTest.java (hunk 1 above)
Comment (raphw) [reply src/test/java/org/mockito/internal/creation/bytebuddy/SubclassByteBuddyMockMakerTest.java]: Fixed the test!

**Review by raphw — APPROVED** [review]

**Pull request merged**